src/eradiate/scenes/illumination/_spot.py

```diff
@@ -0,0 +1,212 @@
+        attrs.field(converter=Path),
+        type="Path",
+        init_type="path-like",
+        default=None,
+        doc="Path to the beam profile file, relative to `path`. Must be a valid image file.",
+    )
+
+    @classmethod
+    def from_size(
+        cls,
+        target: np.typing.ArrayLike,
+        direction: np.typing.ArrayLike,
+        spot_radius: pint.Quantity,
+        beam_divergence: pint.Quantity,
+        **kwargs,
+    ) -> SpotIllumination:
+        """
+        Define the spot illumination through a target spot size and beam width.
+
+        The origin distance is inferred from the requested beam width and spot size.
+
+        Parameters:
+        -----------
+
+        target : array-like
+            Target point for the spot. Unitless values are converted to ``ucc['length']``
+
+        direction : array-like
+            Pointing direction for the spot.
+
+        spot_radius : float or quantity
+            Radius of the desired spot at the target position.
+            Unitless values are converted to ``ucc['length']``
+
+        beam_divergence : float or unit
+            Divergence angle of the spot.
+            Unitless values are convertedd to ``ucc['angle']``
+
+        beam_profile : path-like
+            Image file to define a transversal profile for the beam.
+            Bitmap file.
+
+        **kwargs
+            Remaining keyword arguments are forwarded to the
+            :class:`.SpotIllumination` constructor.
+
+        Returns
+        -------
+        SpotIllumination
+        """
+        target = pinttr.util.ensure_units(target, default_units=ucc.get("length"))
+        half_angle = (
+            pinttr.util.ensure_units(beam_divergence, default_units=ureg.radian) / 2.0
+        )
+        tan_divergence = np.tan(half_angle)
+        distance = spot_radius / tan_divergence
+
+        origin = target - (direction * distance)
+
+        if not any(np.cross(origin.magnitude, [0, 0, 1])):
+            up = [0, 1, 0]
+        else:
+            up = [0, 0, 1]
+
+        return cls(
+            origin=origin, target=target, up=up, beam_width=beam_divergence, **kwargs
+        )
+
+    def kernel_dict(self, ctx: KernelDictContext) -> KernelDict:
+        target = self.target.m_as(uck.get("length"))
+        origin = self.origin.m_as(uck.get("length"))
+
+        retdict = {
+            self.id: {
+                "type": "spot",
+                "to_world": mi.ScalarTransform4f.look_at(
+                    origin=origin, target=target, up=self.up
+                ),
+                "beam_width": self.beam_width.m_as(uck.get("angle")),
+                "cutoff_angle": self.beam_width.m_as(uck.get("angle")),
+                "intensity": self.intensity.kernel_dict(ctx=ctx)["spectrum"],
+            }
+        }
+
+        if self.beam_profile is not None:
+            retdict[self.id]["texture"] = {
+                "type": "bitmap",
+                "filename": self.beam_profile,
```

Comment:
```suggestion
                "filename": str(self.beam_profile),
```
otherwise we will get a `pathlib.PosixPath` in the dictionary, which mistuba does not like iirc.